### PR TITLE
Feature/network request

### PR DESCRIPTION
**Pull request by reneepinna**
What I did

-  Add a function to GET movie data
-  Replace all dummy data with date from api calls
-  Add prop types to components that receive props for better error messaging.

### CODE REVIEW

File: src/apiCalls.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+ function getMovies(id) {
+  return fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id || ''}`)
```
Comment (Jwensloff) [inline src/apiCalls.js:2]: I'm so stoked we were able to refactor to this!

File: src/Components/App/App.js [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
   };
 
- const backToHomePage = () => {
-  setMovieDetails(false);
- }
- const error = true;
+  const backToHomePage = () => {
+    setMovieDetails(false);
+  };
+  const error = false;
```
Comment (nicktu12) [inline src/Components/App/App.js:24]: Could this error flag be stored in state? Kind of odd in react to have a floating variable like this.

File: src/Components/MovieDetail/MovieDetail.js [hunk 3]
```diff
@@ -1,20 +1,42 @@
 import './MovieDetail.scss'
+import PropTypes, { array, arrayOf } from 'prop-types'
+
 function MovieDetails({ movieDetails, backToHomePage }) {
-  let { id, poster_path, backdrop_path, title, average_rating, release_date } =
+  let { id, poster_path, backdrop_path, title, release_date, overview, genres, budget, revenue, runtime, tagline, average_rating} =
```
Comment (nicktu12) [inline src/Components/MovieDetail/MovieDetail.js:5]: mega-destructing!

File: src/Components/MovieDetail/MovieDetail.js [hunk 4]
```diff
@@ -1,20 +1,42 @@
...
     movieDetails;
   return (
     <section className='movieDetails'>
       <img className='movieDetails__img' src={backdrop_path} />
       <section className='movieDetails__block'>
         <h2 className='movieDetails__title'>{title}</h2>
-        <p className='movieDetails__text'>Tagline</p>
-        <p className='movieDetails__text'>Description</p>
+        <p className='movieDetails__text'>{tagline}</p>
```
Comment (nicktu12) [inline src/Components/MovieDetail/MovieDetail.js:12]: Could there possibly be a way to iteratively generate all these p tags?? 🤔

File: src/apiCalls.js (hunk 1 above)
Comment (nicktu12) [reply src/apiCalls.js:2]: Yes this is cool 😎

File: src/apiCalls.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    .then(resp => resp.json())
+    .then(data => {
+      if (!id) {
+        return data.movies
+      }
+      return data.movie
+    })
```
Comment (nicktu12) [inline src/apiCalls.js:9]: Don't forget your catch block!

**Pull request merged**